examples/asrc_demo/src/rate_server.c

```diff
@@ -172,50 +186,33 @@ static float_s32_t determine_avg_I2S_rate_from_driver()
     return result;
 }
 
-void calc_avg_i2s_send_buffer_level(int current_level, bool reset)
+static inline sw_pll_q24_t get_Kp_for_i2s_buffer_control(int32_t nominal_i2s_rate)
 {
-    static int64_t error_accum = 0;
-    static int32_t count = 0;
-
-    if(reset == true)
+    // The Kp constants are generated using the simulation framework, largely through trial and error, to get values using which
+    // the calculated correction factor stablises the buffer level.
+    sw_pll_q24_t Kp = 0;
+    if(((int)nominal_i2s_rate == (int)44100) || ((int)nominal_i2s_rate == (int)48000))
     {
-        error_accum = 0;
-        count = 0;
-        g_avg_i2s_send_buffer_level = 0;
-        g_prev_avg_i2s_send_buffer_level = 0;
-        rtos_printf("Reset avg I2S send buffer level\n");
+        Kp = SW_PLL_Q24(11.542724608); // 0.000000043 * (2**28)
```

Comment:
Maybe even have these magic (K) numbers brought out to defines somewhere in the app?

Reply:
#define KP_RATIONAL_RATE_RATIO or something